File: 206. Reverse Linked List/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        return reverse(head,null);
    }

    private static ListNode reverse(ListNode head,ListNode acc){
        if(head == null) return acc;
        return reverse(head.next,acc = new ListNode(head.val,acc));
    }
}

File: 206. Reverse Linked List/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverse(self, head, acc) -> Optional[ListNode]:
        if not head:
            return acc
        return self.reverse(head.next, ListNode(head.val,acc))

    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        return self.reverse(head, None)
        